### PR TITLE
slack: Add zerver_usermessage object.

### PR DESCRIPTION
**Pull request by rheaparekh**
We check if the recipients in `zerver_subscription` and` zerver_message` are same and then we append the message `id` from `zerver_message` with the `user_profile` from `zerver_subscription`. `flags_mask` are defaulted to 1.

### CODE REVIEW

**Comment by rht** [comment]
This fixes 2 TODO's.

File: zulip/integrations/slack/slackdata2zulipdata.py [hunk 1]
```diff
@@ -337,11 +336,28 @@ def main(slack_zip_file: str) -> None:
...
+    zerver_usermessage = []
+
     for channel in added_channels.keys():
         zerver_message.append(channelmessage2zerver_message(slack_dir, channel,
                               added_users, added_channels))
+    # construct the usermessage object and append it to zerver_usermessage
+    usermessage_id = 1
+    for zulip_message in zerver_message:
+        for subscription in zerver_subscription:
+            if subscription['recipient'] == zulip_message['recipient']:
```
Comment (rht) [inline zulip/integrations/slack/slackdata2zulipdata.py]: Maybe should add a comment saying that this is a stream message?

File: zulip/integrations/slack/slackdata2zulipdata.py [hunk 2]
```diff
@@ -337,11 +336,28 @@ def main(slack_zip_file: str) -> None:
...
+                usermessage = dict(
+                    user_profile=subscription['user_profile'],
+                    id=usermessage_id,
+                    flags_mask=1,  # defaulting to 'read'
```
Comment (rht) [inline zulip/integrations/slack/slackdata2zulipdata.py]: Now this paves the way to mapping Slack's pins to Zulip's stars!

File: zulip/integrations/slack/slackdata2zulipdata.py (hunk 1 above)
Comment (rht) [reply zulip/integrations/slack/slackdata2zulipdata.py]: Ah, right, it loops over all the zerver_subscription, which includes streams and huddles etc, my mistake!

Comment (rheaparekh) [reply zulip/integrations/slack/slackdata2zulipdata.py]: @rht is this necessary? zerver_message itself is interpreted to be the stream message.

**Comment by rht** [comment]
Merged, awesome @rheaparekh !

**Pull request merged**